File: arm.64/langs/ocaml/src/main.ml
let () =
  let target_dir = if Array.length Sys.argv > 1 then Some Sys.argv.(1) else None in
  let dist = Boot.boot target_dir in
  if dist <> [] then (
    let trips = Trip.trip () in
    List.iter (fun d ->
      List.iter (fun t ->
        ignore (Bind.caml_call_pump (Some d.Type.pump_fn) t.Type.address t.Type.payload t.Type.options)
      ) trips
    ) dist
  );
  Printf.printf "{\n  \"lang\": \"ocaml\",\n  \"status\": \"ready\",\n  \"engines\": %d\n}\n" (List.length dist)

File: arm.64/langs/rust/src/main.rs
mod bind;
mod boot;
mod find;
mod trip;
mod r#type;

use std::env;
use std::ffi::CString;

fn main() {
    let args: Vec<String> = env::args().collect();
    let target_dir = args.get(1).map(|s| s.as_str());

    let dist = boot::boot(target_dir);
    if !dist.is_empty() {
        let trips = trip::trip();
        for d in &dist {
            for t in &trips {
                let addr = CString::new(t.address).unwrap();
                let pay = CString::new(t.payload).unwrap();
                let opt = CString::new(t.options).unwrap();
                unsafe {
                    (d.pump)(addr.as_ptr(), pay.as_ptr(), opt.as_ptr());
                }
            }
        }
    }

    println!("{{\n  \"lang\": \"rust\",\n  \"status\": \"ready\",\n  \"engines\": {}\n}}", dist.len());
}
